core/src/main/java/org/infinispan/partitionhandling/impl/PreferAvailabilityStrategy.java

```diff
@@ -188,6 +196,8 @@ public void onPartitionMerge(AvailabilityStrategyContext context, Map<Address, C
             mergedTopology = new CacheTopology(maxTopologyId + 1, maxRebalanceId + 1, maxTopology.getCurrentCH(),
                   mergehash, mergehash, CacheTopology.Phase.CONFLICT_RESOLUTION, newMembers, persistentUUIDManager.mapAddresses(newMembers));
          } else {
+            // TODO If maxTopology.getPhase() == READ_NEW_WRITE_ALL, the pending CH would be more appropriate
+            // The best approach may be to collect the read owners from all the topologies and use their union as the current CH
```

Comment:
Once my PR gets merged, we will also be doing this for [Conflict resolution](https://github.com/ryanemerson/infinispan/blob/1eed144ffe530d739dda2bc5f25ed0679b2185d0/core/src/main/java/org/infinispan/partitionhandling/impl/PreferAvailabilityStrategy.java#L199-L201) as we discussed